### PR TITLE
[NUXE-269] KIT DEV - IconStatValue kit

### PR DESCRIPTION
**Pull request by idanzonshein**
#### Github Issue URL

https://github.com/powerhome/playbook/issues/999

#### Screens

Screen Recording 2020-11-17 at 10 43 42 AM

#### Breaking Changes

Only additive changes.

#### Checklist:

- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review
- [x] **SCREENSHOT** Please add a screen shot or two
- [x] **SPECS** Please cover your changes with specs

### CODE REVIEW

File: app/pb_kits/playbook/pb_icon_stat_value/_icon_stat_value.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
+@import "../pb_title/title";
+@import "../pb_caption/caption";
+@import "../tokens/spacing";
+@import "../pb_icon_circle/icon_circle";
+
+
+
+
```
Comment (eabousaif) [inline app/pb_kits/playbook/pb_icon_stat_value/_icon_stat_value.scss:8]: Slight nitpick to clear out some of the empty lines here

**Review by eabousaif — APPROVED** [review]

File: app/pb_kits/playbook/_playbook.scss [hunk 2]
```diff
@@ -81,3 +81,5 @@
 @import 'pb_time_stacked/time_stacked';
 @import 'pb_weekday_stacked/weekday_stacked';
 @import './utilities/spacing';
+
+@import 'pb_icon_stat_value/icon_stat_value';
```
Comment (garettarrowood) [inline app/pb_kits/playbook/_playbook.scss:85]: This should be moved into alphabetical order above.

File: app/pb_kits/playbook/data/menu.yml [hunk 3]
```diff
@@ -93,3 +94,5 @@ kits:
     - weekday_stacked
   - user
   - user_badge
+
+
```
Comment (garettarrowood) [inline app/pb_kits/playbook/data/menu.yml:98]: Extra lines

File: app/pb_kits/playbook/packs/examples.js [hunk 4]
```diff
@@ -172,4 +174,3 @@ WebpackerReact.setup({
   ...UserBadge,
   ...WeekdayStacked,
 })
-
```
Comment (garettarrowood) [inline app/pb_kits/playbook/packs/examples.js:175]: Leave this extra line

File: app/pb_kits/playbook/pb_icon_stat_value/_icon_stat_value.html.erb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        <%= pb_rails "title", props: {
+            text: object.value_unit,
+            size: object.title_size } %>
+
+        <%= pb_rails "caption", props: {
+            text: object.text } %>
+
+      </div>
+
+
```
Comment (garettarrowood) [inline app/pb_kits/playbook/pb_icon_stat_value/_icon_stat_value.html.erb:22]: Extra lines

File: app/pb_kits/playbook/pb_icon_stat_value/_icon_stat_value.scss [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      justify-content: flex-end;
+    }
+
+    [class^=pb_icon_circle] {
+      margin-right: $space-sm;
+    }
+  }
+
+
+
```
Comment (garettarrowood) [inline app/pb_kits/playbook/pb_icon_stat_value/_icon_stat_value.scss:56]: ditto

**Review by garettarrowood — CHANGES_REQUESTED** [review]
Great work on this! Only nit pick feedback from me.

**Comment by benlangfeld** [comment]
Please merge master into this before adding the milano label back - the resource allocations on this branch are wrong and are alerting for quota exhaustion. /cc @iblackman

**Comment by garettarrowood** [comment]
@benlangfeld -  I don't fully understand how deployment environments are used in playbook. Meaning, since we don't notify ninjas about testing these changes in playbook, who is using the deployment? Does it need to be there at all? Do we add the tag just to verify that things do deploy properly?  cc/ @thestephenmarshall

**Comment by idanzonshein** [comment]
@benlangfeld Updated.

**Review by garettarrowood — APPROVED** [review]
LGTM

File: app/pb_kits/playbook/_playbook.scss [hunk 7]
```diff
@@ -81,4 +82,4 @@
 @import 'pb_user_badge/user_badge';
 @import 'pb_time_stacked/time_stacked';
 @import 'pb_weekday_stacked/weekday_stacked';
-@import './utilities/spacing';
+@import './utilities/spacing';
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/_playbook.scss:85]:
```suggestion
@import './utilities/spacing';

```

We should not remove the extra line.

File: app/pb_kits/playbook/data/menu.yml [hunk 8]
```diff
@@ -94,4 +95,4 @@ kits:
     - title_detail
     - weekday_stacked
   - user
-  - user_badge
+  - user_badge
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/data/menu.yml:98]:
```suggestion
  - user_badge
  
```

File: app/pb_kits/playbook/index.js [hunk 9]
```diff
@@ -34,6 +34,7 @@ export HomeAddressStreet from './pb_home_address_street/_home_address_street.jsx
 export Icon from './pb_icon/_icon.jsx'
 export IconCircle from './pb_icon_circle/_icon_circle.jsx'
 export IconValue from './pb_icon_value/_icon_value.jsx'
+export IconStatValue from 'pb_icon_stat_value/_icon_stat_value.jsx'
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/index.js:37]:
```suggestion
export IconStatValue from 'pb_icon_stat_value/_icon_stat_value.jsx'
export IconValue from './pb_icon_value/_icon_value.jsx'
```

These are not alpha ordered.

File: app/pb_kits/playbook/packs/examples.js [hunk 10]
```diff
@@ -47,6 +47,7 @@ import * as HomeAddressStreet from 'pb_home_address_street/docs'
 import * as Icon from 'pb_icon/docs'
 import * as IconCircle from 'pb_icon_circle/docs'
 import * as IconValue from 'pb_icon_value/docs'
+import * as IconStatValue from 'pb_icon_stat_value/docs'
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/packs/examples.js:50]:
```suggestion
import * as IconStatValue from 'pb_icon_stat_value/docs'
import * as IconValue from 'pb_icon_value/docs'
```

Alpha ordering

File: app/pb_kits/playbook/packs/examples.js [hunk 11]
```diff
@@ -129,6 +130,7 @@ WebpackerReact.setup({
   ...Icon,
   ...IconCircle,
   ...IconValue,
+  ...IconStatValue,
```
Comment (thestephenmarshall) [inline app/pb_kits/playbook/packs/examples.js:133]: Alpha

File: spec/pb_kits/playbook/kits/icon_stat_value_spec.rb [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+                      .with_default("sm")
+                      .with_values("sm", "lg") }
+  it { is_expected.to define_enum_prop(:variant)
+                      .with_default("default")
+                      .with_values("default", "royal", "blue", "purple", "teal", "red", "yellow", "green") }
+
+  describe "#classname" do
+    it "returns namespaced class name", :aggregate_failures do
+      expect(subject.new({}).classname).to eq "pb_icon_stat_value_kit_horizontal_sm_default"
+      expect(subject.new(orientation: "vertical", size:"lg", variant: "royal").classname).to eq "pb_icon_stat_value_kit_vertical_lg_royal"
```
Comment (thestephenmarshall) [inline spec/pb_kits/playbook/kits/icon_stat_value_spec.rb:20]: This is missing the other test cases which cover the other variants.

**Review by thestephenmarshall — CHANGES_REQUESTED** [review]
Nice work so far! Only a few nits (see the suggestions) and I think the spec for this kits needs a little more coverage.

**Comment by jasperfurniss** [comment]
Just removed the milano label because milano is at capacity. See the `Nitro Release Engineering` channel in Connect for more.

**Comment by garettarrowood** [comment]
@jasperfurniss - I had a brief convo with @benlangfeld about Playbook milano this morning to gain some clarity for myself and the Champions. Here is what he had to say 

Ben: 
"... playbook doesn't use nearly as much in CPU/memory as nitro-web does. You can compare [the nitro-web quota](https://github.com/powerhome/APP/blob/master/application-namespaces/tier/beta/nitro-web-prs.yaml#L41-L60) to [the playbook quota](https://github.com/powerhome/APP/blob/master/application-namespaces/tier/beta/playbook-prs.yaml#L41-L60). Note that they are different pools. The [limit on number of nitro-web review stacks per cluster](https://github.com/powerhome/APP/blob/master/application-namespaces/tier/beta/nitro-web-prs.yaml#L6) is distinct from [that for playbook](https://github.com/powerhome/APP/blob/master/application-namespaces/tier/beta/playbook-prs.yaml#L6)."

So in other words - `nitro-web` and `playbook` deployments do not share the same resources. So removing the `milano` label here will not affect the limit reached that @rafbgarcia posted in the `Nitro Release Engineering` room in Connect.

**Review by garettarrowood — APPROVED** [review]

File: app/pb_kits/playbook/index.js [hunk 13]
```diff
@@ -33,6 +33,7 @@ export Highlight from './pb_highlight/_highlight.jsx'
 export HomeAddressStreet from './pb_home_address_street/_home_address_street.jsx'
 export Icon from './pb_icon/_icon.jsx'
 export IconCircle from './pb_icon_circle/_icon_circle.jsx'
+export IconStatValue from 'pb_icon_stat_value/_icon_stat_value.jsx'
```
Comment (coleerikson) [inline app/pb_kits/playbook/index.js:36]: Is this missing a `./`?

File: app/pb_kits/playbook/pb_icon_stat_value/icon_stat_value.rb [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+                     values: %w[default royal blue purple teal red yellow green],
+                     default: "default"
+
+      prop :orientation, type: Playbook::Props::Enum,
+                         values: %w[vertical horizontal],
+                         default: "horizontal"
+      prop :unit
+      prop :text
+      prop :value
+
```
Comment (coleerikson) [inline app/pb_kits/playbook/pb_icon_stat_value/icon_stat_value.rb:26]: Can any of these "unit, text, value" support validation types like "icon, size, variant, orientation"?

**Review by coleerikson — APPROVED** [review]
LGTM, had a couple comments.

File: app/pb_kits/playbook/index.js (hunk 13 above)
Comment (coleerikson) [reply app/pb_kits/playbook/index.js:36]: Can confirm. This breaks in webpacker.

Comment (coleerikson) [reply app/pb_kits/playbook/index.js:36]: Can confirm. This breaks in webpacker.

**Review by coleerikson — CHANGES_REQUESTED** [review]

**Review by coleerikson — APPROVED** [review]

**Comment by thestephenmarshall** [comment]
@garettarrowood sorry I missed your comment amongst others. So we do want to ensure the deployment completes. As of right now, if we want ninja testing on a PR here we do manually notify someone. We use the deployment env ourselves to ensure Playbook is operating as expected within the context of a given PR. This prevents us from having to constantly re-install locally and deal with the entropy for each given PR (which, for a UXer responsible for the release, could be upwards of 15 total). Hope that helps.

**Review by thestephenmarshall — APPROVED** [review]

**Comment by garettarrowood** [comment]
@thestephenmarshall - It does. Thanks!

**Comment by garettarrowood** [comment]
Woooohoooo!!!
